Add init design doc method and make tests run as is

phil1618 opened the pull request.

phil1618 committed:
Merge remote-tracking branch 'origin/master' into update-jackson-depencies

phil1618 committed:
Add initStandardDesignDocument

phil1618 committed:
Add Design Doc initialization block in tests

aduchate commented on src/main/kotlin/org/taktik/couchdb/Client.kt:
```diff
@@ -254,6 +258,20 @@ class ClientImpl(private val httpClient: WebClient,
+                return this.update(existingDesignDocument.let { merged.copy(rev = it.rev) })
+            }
+
+            return existingDesignDocument
```
C'est plus kotlin d'avoir un seul return et de tirer parti du fait que if est une expression

aduchate commented on src/test/kotlin/org/taktik/couchdb/Code.kt:
```diff
@@ -20,11 +20,15 @@ import com.fasterxml.jackson.annotation.JsonIgnoreProperties
 @JsonInclude(JsonInclude.Include.NON_NULL)
 @JsonIgnoreProperties(ignoreUnknown = true)
 @KotlinBuilder
+@Views(View(name = "all", map = "function(doc) { if (doc.java_type == 'Code' && !doc.deleted) emit( null, doc._id )}"),
```
C'est un peu bizarre de dévoyer Code du statut de domain object à DAO. Ce code pourrait servir d'exemple et j'ai peur que les gens ne comprennent pas.

aduchate commented on src/test/kotlin/org/taktik/couchdb/CouchDbClientTests.kt:
```diff
@@ -69,6 +70,18 @@ class CouchDbClientTests {
+                client.create(8, 2)
+            }
+
+            val codeDesignDocument = StdDesignDocumentFactory().generateFrom("_design/Code", Code.from("test", UUID.randomUUID().toString(), "test"))
```
En plus... Au lieu de se retrouver à devoir initialiser les documents sur base d'un singleton, tu dois artificiellement créer l'objet

phil1618 commented:
@aduchate Tu as raison. La facilité d'exécution des tests ne justifie pas la simplification de l'absence de notion de DAO. Je vais donc revoir ma copie.

phil1618 committed:
Setup database for tests and add DAO example

Fix typo


Make use of js view file definition


Provide default system properties


Fix testings value change


Introduce DAO interfacing


Cleaning 


Cleaning

phil1618 committed:
Cleaning
